src/Uno.UWP/Buffers/LinearArrayPool.cs

```diff
@@ -0,0 +1,224 @@
+#nullable enable
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Windows.System;
+
+namespace Uno.Buffers
+{
+	internal class LinearArrayPool<T>
+	{
+		private readonly int _baseArraySize;
+
+		private readonly Bucket[] _buckets;
+
+		private int _requestTrimCallbackRegistration;
+
+		private static readonly bool _automaticMemoryManagement;
+		private static readonly bool _canUseMemoryManager;
+		private static readonly int _defaultAutomaticBucketSize;
+		private static readonly int _defaultBucketSize;
+
+		private static IArrayPoolPlatformProvider _platformProvider = new DefaultArrayPoolPlatformProvider();
+
+		static LinearArrayPool()
+		{
+			_automaticMemoryManagement = WinRTFeatureConfiguration.ArrayPool.EnableAutomaticMemoryManagement;
+
+			_canUseMemoryManager = _automaticMemoryManagement && _platformProvider.CanUseMemoryManager;
+
+			_defaultAutomaticBucketSize = WinRTFeatureConfiguration.ArrayPool.DefaultAutomaticMaxNumberOfArraysPerBucket;
+			_defaultBucketSize = WinRTFeatureConfiguration.ArrayPool.DefaultMaxNumberOfArraysPerBucket;
```

Comment:
Can we inline these and avoid the static constructor?

Reply:
There will be a static constructor anyways, it probably won't change much.